internal/serialization/generic_compact.go

```diff
@@ -0,0 +1,131 @@
+package serialization
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+type compactFieldReader func(r serialization.CompactReader, field string) any
+
+type compactFieldWriter func(w serialization.CompactWriter, field string, value any)
+
+type GenericCompact struct {
+	ValueType     reflect.Type
+	ValueTypeName string
+	Fields        []CompactField
+	readers       []compactFieldReader
+	writers       []compactFieldWriter
+}
+
+func NewGenericCompact(value GenericCompact) (*GenericCompact, error) {
+	rs := make([]compactFieldReader, len(value.Fields))
+	ws := make([]compactFieldWriter, len(value.Fields))
+	for i, f := range value.Fields {
+		if f.Type < CompactFieldType(serialization.FieldKindNotAvailable) || f.Type > CompactFieldType(serialization.FieldKindArrayOfNullableFloat64) {
+			return nil, fmt.Errorf("invalid portable type: %d", f.Type)
+		}
+		r, ok := compactReaders[serialization.FieldKind(f.Type)]
+		if !ok {
+			return nil, fmt.Errorf("reader not found for compact type: %d", f.Type)
+		}
+		rs[i] = r
+		// writing is disabled for now --YT
+	}
+	return &GenericCompact{
+		Fields:        value.Fields,
+		ValueType:     value.makeType(),
+		ValueTypeName: value.ValueTypeName,
+		readers:       rs,
+		writers:       ws,
+	}, nil
+
+}
+
+func (cm GenericCompact) makeType() reflect.Type {
+	fs := make([]reflect.StructField, len(cm.Fields))
+	for i, f := range cm.Fields {
+		fs[i] = reflect.StructField{
+			Name: fmt.Sprintf("Field%03d", i),
+			Type: fieldKindToType[serialization.FieldKind(f.Type)],
+			Tag:  reflect.StructTag(fmt.Sprintf("json:\"%s\"", f.Name)),
+		}
+	}
+	return reflect.StructOf(fs)
+}
+
+func (cm GenericCompact) Type() reflect.Type {
+	return cm.ValueType
+}
+
+func (cm GenericCompact) TypeName() string {
+	return cm.ValueTypeName
+}
+
+func (cm GenericCompact) Read(reader serialization.CompactReader) interface{} {
+	rs := cm.readers
+	v := reflect.New(cm.ValueType)
+	for i, f := range cm.Fields {
+		value := reflect.ValueOf(rs[i](reader, f.Name))
+		if value.Interface() == nil {
+			continue
+		}
+		v.Elem().Field(i).Set(value)
+	}
+	return v.Interface()
+}
+
+func (cm GenericCompact) Write(writer serialization.CompactWriter, value interface{}) {
+	// TODO: implement me
+	panic("implement me")
```

Comment:
More revealing explanation can be given like `write operation for compact serialization is not supported yet.`

Reply:
That code is never executed by CLC. I think it's purpose would be probably understood by developers.